src/ucp/tag/offload.c

```diff
@@ -43,11 +44,30 @@ void ucp_tag_offload_completed(uct_tag_context_t *self, uct_tag_t stag,
         /* Sync send - need to send a reply */
         ucp_tag_offload_eager_sync_send_ack(req->recv.worker, imm, stag);
     }
-    ucp_request_memory_dereg(ctx, iface->rsc_index, req->recv.datatype,
-                             &req->recv.state);
+
+    if (req->flags & UCP_REQUEST_FLAG_OFFLOADED_BB) {
```

Comment:
can we use req->send.reg_rsc from PR #1574 to indicate if need to unregister memory, instead of introducing UCP_REQUEST_FLAG_OFFLOADED_BB flag?